### PR TITLE
Add support for file and process-related syscalls

### PR DESCRIPTION
**Pull request by rvolosatovs**
Part of #34

### CODE REVIEW

**Review by haraldh — CHANGES_REQUESTED** [review]
oops.. approved too soon.

File: src/v2/src/guest/handler.rs [hunk 1]
```diff
@@ -453,4 +542,24 @@ impl<'a, P: Platform> Execute for Handler<'a, P> {
...
+        if signum >= sigrtmax || sigsetsize != 8 {
+            return Err(libc::EINVAL);
+        }
+        if let Some(oldact) = oldact {
+            *oldact = self.last_action;
+        }
+        if let Some(act) = act {
+            self.last_action = Some(*act);
+        }
+        Ok(())
```
Comment (haraldh) [inline src/v2/src/guest/handler.rs:563]: For every `signum` there is a stored old pointer to be returned... so use an array, as in the original

**Review by haraldh — CHANGES_REQUESTED** [review]

File: src/v2/src/guest/handler.rs [hunk 2]
```diff
@@ -264,10 +294,14 @@ pub trait Execute {
...
 }
 
+const SIGRTMAX: c_int = 64;
+
 /// Guest request handler.
 pub struct Handler<'a, P: Platform> {
     alloc: Alloc<'a, phase::Init>,
     platform: P,
+
+    actions: [Option<sigaction>; SIGRTMAX as _],
```
Comment (haraldh) [inline src/v2/src/guest/handler.rs:315]: We should create an issue, that if we ever support more than one thread, the `Handler` should be thread local then.

File: src/v2/src/guest/handler.rs [hunk 3]
```diff
@@ -276,6 +310,7 @@ impl<'a, P: Platform> Handler<'a, P> {
         Self {
             alloc: Alloc::new(block),
             platform,
+            actions: [None; SIGRTMAX as _],
```
Comment (haraldh) [inline src/v2/src/guest/handler.rs:324]: So, is this `static` ???

File: src/v2/src/guest/handler.rs (hunk 2 above)
Comment (rvolosatovs) [reply src/v2/src/guest/handler.rs:315]: Potentially, we could take this in as a parameter and share between handlers

**Review by haraldh — CHANGES_REQUESTED** [review]

File: src/v2/src/guest/handler.rs (hunk 3 above)
Comment (rvolosatovs) [reply src/v2/src/guest/handler.rs:324]: No, why would it need to be?

**Review by haraldh — APPROVED** [review]

**Pull request merged**